nowcasting_dataset/data_sources/nwp/nwp_data_source.py

```diff
@@ -131,7 +140,10 @@ def datetime_index(self) -> pd.DatetimeIndex:
             nwp = self._open_data()
         else:
             nwp = self._data
-        target_times = nwp["init_time"] + nwp["step"][:3]
+        # We need to return the `target_times` (the times the NWPs are _about_).
+        # The `target_time` is the `init_time` plus the forecast horizon `step`.
+        # `step` is an array of timedeltas, so we can just add `init_time` to `step`.
+        target_times = nwp["init_time"] + nwp["step"]
```

Comment:
I've removed the `[:3]` so that we don't just use the first 3 steps (which would produce gaps in the target_datetimes if any NWP inits are missing)